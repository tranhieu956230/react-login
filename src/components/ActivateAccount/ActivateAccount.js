import React, { useEffect, useState } from "react";
import styled from "styled-components";

import { activateAccount } from "services";

const ActivateAccount = props => {
  const [isActivated, setIsActivated] = useState(false);

  useEffect(() => {
    activateAccount(props.match.params.code).then(response => {
      if (response.code === "SUCCESS") {
        setIsActivated(true);
      }
      props.history.push("/");
    });
  }, []);

  return (
    <Text data-test={"container"}>
      {isActivated
        ? "Your account is activated. Please log in"
        : "Your account is being activated"}
    </Text>
  );
};

const Text = styled.h1`
  font-size: 5rem;
  font-weight: 500;
  text-align: center;
  margin-top: 2rem;
  color: rgba(41, 41, 41, 0.82);
`;

export default ActivateAccount;
