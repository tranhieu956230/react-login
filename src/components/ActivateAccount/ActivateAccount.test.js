import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter, Route } from "react-router-dom";

import ActivateAccount from "./index";

jest.mock("../../services/index.js");

describe("Test Activate Account Component", () => {
  it("Should render with no error", () => {
    const wrapper = shallow(<ActivateAccount />);
    const text = wrapper.find("h1").text();
    expect(text).toEqual("We are activating your account");
  });

  it("Should match the snapshot", () => {
    const tree = renderer.create(<ActivateAccount />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render activated", async done => {
    const code = {
      match: {
        params: {
          code: 1000
        }
      }
    };

    const wrapper = mount(
      <BrowserRouter>
        <Route render={props => <ActivateAccount {...props} {...code} />} />
      </BrowserRouter>
    );
    setTimeout(() => {
      const text = wrapper.find("h1").text();
      expect(text).toEqual("Your account is activated. Please log in");
      done();
    });
  });
});
