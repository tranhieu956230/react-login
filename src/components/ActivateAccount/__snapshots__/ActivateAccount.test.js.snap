// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ActivateAccount Shallow Should match the snapshot 1`] = `
<h1
  className="activate-account__title"
>
  We are activating your account
</h1>
`;
